### PR TITLE
CXM-67496 Support for US gov 

### PR DESCRIPTION
**Pull request by tirupatibg**
### Support for the US gov

Current hazelcast-azure library doesn't have an option for us gov deployement, because of this we have created the custom strategy for the US-Gov com.hazelcast.azure.gov.us.AzureGovUSDiscoveryStrategy and this new strategy supported class files are in the new customer jar file hazelcast-azure-gov-us.1.0.jar

### Here are the changes I have made for this PR

1. Added the new jar file to the path and made an entry in the azuredeploy.json to copy this jar file to the target machine 
1. Add the new parameter named deployementType having support two values commercial or us-gov 
1. Modified the bootstrap file to incorporate this new param and to transfer this param to new modify configuration file so that we can define the strategy in hazelcast.xml based on deployement type
1. In modifyconfiguation file, based on deployement type put the respective strategy
1. In the install_hazelcast, added the new folder for the hz-mvn repo to store the new lib
1. And in pom.xml, added the new local repo and added the jar file dependency

### CODE REVIEW

File: hazelcast-vm-cluster/azuredeploy.json [hunk 1]
```diff
@@ -143,9 +143,20 @@
...
+            ],
+            "defaultValue": "commercial",
+            "type": "String",
+            "metadata": {
+                "description": "the deployement type used to identify the deployement mode either gov or commercial"
+            }
+        },        
         "templateBaseUrl": {
             "type": "string",
-            "defaultValue": "https://raw.githubusercontent.com/srinusanchula/azure-quickstart-templates/master/hazelcast-vm-cluster/",
```
Comment (srinusanchula) [inline hazelcast-vm-cluster/azuredeploy.json]: Please move the references back to srinusanchula repo.

File: hazelcast-vm-cluster/install_hazelcast [hunk 2]
```diff
@@ -5,8 +5,10 @@ sudo apt install maven -y
 
 # Copy configuration files and install
 sudo mkdir /var/hazelcast
+sudo mkdir -p /var/hz/mvn-repo
```
Comment (srinusanchula) [inline hazelcast-vm-cluster/install_hazelcast]: Can we create local repo 'mvn-repo' under /var/hazelcast?

/var/hazelcast is the path where we wanted to put all our files.

File: hazelcast-vm-cluster/azuredeploy.json (hunk 1 above)
Comment (tirupatibg) [reply hazelcast-vm-cluster/azuredeploy.json]: Yes, I did. Once I validate with my branch. I reverted back to srinusanchula repo

File: hazelcast-vm-cluster/install_hazelcast [hunk 3]
```diff
@@ -5,8 +5,10 @@ sudo apt install maven -y
...
 sudo cp mancenter_hazelcast /var/hazelcast
 chmod +x /var/hazelcast/mancenter_hazelcast
 sudo cp pom.xml /var/hazelcast
+sudo mvn deploy:deploy-file -Dfile=hazelcast-azure-gov-us-1.0.jar -DgroupId=com.hazelcast -DartifactId=hazelcast-azure-gov-us -Dversion=1.0 -Durl=file:/var/hz/mvn-repo/ -Dpackaging=jar -DrepositoryId=maven-repository -DupdateReleaseInfo=true
```
Comment (srinusanchula) [inline hazelcast-vm-cluster/install_hazelcast]: If you change above, you need to change it too.

File: hazelcast-vm-cluster/azuredeploy.json (hunk 1 above)
Comment (tirupatibg) [reply hazelcast-vm-cluster/azuredeploy.json]: Yes, I did. Once I validate with my branch. I reverted back to srinusanchula repo

File: hazelcast-vm-cluster/pom.xml [hunk 4]
```diff
@@ -10,7 +10,19 @@
   <version>1.0</version>
   <name>Hazelcast Grid</name>
 
+  <repositories>
+    <repository>
+      <id>maven-repository</id>
+      <url>file:///var/hz/mvn-repo</url>
```
Comment (srinusanchula) [inline hazelcast-vm-cluster/pom.xml]: Repo path change.

File: hazelcast-vm-cluster/install_hazelcast (hunk 2 above)
Comment (tirupatibg) [reply hazelcast-vm-cluster/install_hazelcast]: Done. Changed the repo path to /var/hazelcast/mvn-repo

File: hazelcast-vm-cluster/install_hazelcast (hunk 3 above)
Comment (tirupatibg) [reply hazelcast-vm-cluster/install_hazelcast]: Done. Changed the repo path to /var/hazelcast/mvn-repo

File: hazelcast-vm-cluster/pom.xml (hunk 4 above)
Comment (tirupatibg) [reply hazelcast-vm-cluster/pom.xml]: Done. Changed the repo path to /var/hazelcast/mvn-repo

**Pull request merged**